contrib/inventory/ec2.py

```diff
@@ -475,6 +479,13 @@ def connect_to_aws(self, module, region):
             connect_args['profile_name'] = self.boto_profile
             self.boto_fix_security_token_in_profile(connect_args)
 
+        if self.iam_role:
```

Comment:
This needs a guard, since `self.iam_role` will be unset if the option isn't specified.